### PR TITLE
Fleshes out smoking content

### PR DESCRIPTION
**Pull request by SweptWasTaken**
## About the PR 

This PR is using mostly cev-eris sprites. I've ported the tg sprites as well but I chose cev-eris because I think it looks better.

- cigarette packs
- cigarette cartons
- cigars and cigar cases
- rollies and rolling paper - I plan to add crafting recipes for different spliffs to this PR. Further down the line we'll have a system for actual reagents in the smoke
- smoking pipe

I've also done content stuff like filling up the cigarette vending machine, adding cigars and cigs to heads lockers and adding a cigarette supply crate.

PR is still a draft as I want to add a few more things.

**Screenshots**

### CODE REVIEW

**Comment by SweptWasTaken** [comment]
Ready for review. Do not research the hoops I had to jump through to get crafting the cigarettes working, but if you do, please look upon it as a catalyst for change to both constructionGraphs and Stack enums. #3314 and #2690 respectively.

Also #3316.

**Comment by Zumorica** [comment]
> Ready for review. Do not research the hoops I had to jump through to get crafting the cigarettes working, but if you do, please look upon it as a catalyst for change to both constructionGraphs and Stack enums. #3314 and #2690 respectively.
> 
> Also #3316.
> 
> PS if a maintainer could please nuke the RobustToolbox conflicting file i'd be grateful. No idea how it got in there.

You could have just used component/prototype steps... This is an awful solution

**Comment by SweptWasTaken** [comment]
> 
> 
> > Ready for review. Do not research the hoops I had to jump through to get crafting the cigarettes working, but if you do, please look upon it as a catalyst for change to both constructionGraphs and Stack enums. #3314 and #2690 respectively.
> > Also #3316.
> > PS if a maintainer could please nuke the RobustToolbox conflicting file i'd be grateful. No idea how it got in there.
> 
> You could have just used component/prototype steps... This is an awful solution

Fixed recipes, now using prototype steps.

**Review by DrSmugleaf — APPROVED** [review]
Fix CigPackBase's sprite component and it's good to merge.
@Zumorica on the botany changes.

File: Resources/Prototypes/Recipes/Crafting/smokeables.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
+﻿- type: construction
```
Comment (metalgearsloth) [inline Resources/Prototypes/Recipes/Crafting/smokeables.yml:1]: This file's exploding in tests

**Comment by metalgearsloth** [comment]
What's with all the empty pngs? Maybe I should give it an actual review...

File: Content.Shared/GameObjects/Components/SharedStackComponent.cs [hunk 2]
```diff
@@ -137,6 +137,12 @@ public enum StackType
...
         FloorTileDark,
         FloorTileWood,
-        MetalRod
+        MetalRod,
+        PaperRolling,
+        CigaretteFilter,
+        GroundTobacco,
+        GroundCannabis,
+        LeavesTobaccoDried,
+        LeavesCannabisDried
```
Comment (Zumorica) [inline Content.Shared/GameObjects/Components/SharedStackComponent.cs:146]: These can now be removed imo

File: Resources/Prototypes/Recipes/Crafting/Graphs/smokeables.yml [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+               icon:
+                 sprite: Objects/Specific/Hydroponics/tobacco.rsi
+                 state: dried
+               name: dried tobacco leaves
+          -  - prototype: LeavesTobaccoDried
+               icon:
+                 sprite: Objects/Specific/Hydroponics/tobacco.rsi
+                 state: dried
+               name: dried tobacco leaves
+               doAfter: 5
```
Comment (Zumorica) [inline Resources/Prototypes/Recipes/Crafting/Graphs/smokeables.yml:97]: The indentation here is wrong, you're using two `-`

**Review by Zumorica — CHANGES_REQUESTED** [review]

File: Resources/Prototypes/Entities/Objects/Consumable/Smokeables/Cigarettes/packs.yml [hunk 4]
```diff
@@ -0,0 +1,87 @@
+- type: entity
+  id: CigPackBase
+  parent: BaseItem
+  name: cigarette pack
+  abstract: true
+  components:
+    - type: Sprite
+      state: closed
```
Comment (Ygg01) [inline Resources/Prototypes/Entities/Objects/Consumable/Smokeables/Cigarettes/packs.yml:8]:
```suggestion
      state: closed
      netsync: false
```

This should solve your issues with cigars not closing. Open/Close bag visualizations are client only.

File: Content.Shared/GameObjects/Components/SharedStackComponent.cs (hunk 2 above)
Comment (SweptWasTaken) [reply Content.Shared/GameObjects/Components/SharedStackComponent.cs:146]: Nah, I still want these to be stacks so they have to be there.

File: Resources/Prototypes/Recipes/Crafting/Graphs/smokeables.yml (hunk 3 above)
Comment (SweptWasTaken) [reply Resources/Prototypes/Recipes/Crafting/Graphs/smokeables.yml:97]: Fixed.

**Comment by SweptWasTaken** [comment]
> 
> 
> What's with all the empty pngs? Maybe I should give it an actual review...

What do you mean? I checked through all the ones I added and none are empty.

**Comment by metalgearsloth** [comment]
Maybe it's github being shithouse?

image

**Comment by SweptWasTaken** [comment]
> 
> 
> Maybe it's github being shithouse?
> 
> image

Yep, that file is 

> 
> 
> Maybe it's github being shithouse?
> 
> image

Yes, ready for a merge.

**Comment by SweptWasTaken** [comment]
Merged master.

**Pull request merged**